app/frontend/src/features/profile/view/UserPage.tsx

```diff
@@ -0,0 +1,90 @@
+import { CircularProgress, Collapse } from "@material-ui/core";
+import Alert from "components/Alert";
+import SuccessSnackbar from "components/SuccessSnackbar";
+import { SEND_REQUEST_SUCCESS } from "features/constants";
+import NewHostRequest from "features/messages/requests/NewHostRequest";
+import { ProfileUserProvider } from "features/profile/hooks/useProfileUser";
+import Overview from "features/profile/view/Overview";
+import UserCard from "features/user/UserCard";
+// import References from "features/profile/view/References";
+import useUserByUsername from "features/userQueries/useUserByUsername";
+import { useLayoutEffect, useState } from "react";
+import { useHistory, useParams } from "react-router-dom";
+import { routeToUser } from "routes";
+import makeStyles from "utils/makeStyles";
+
+const useStyles = makeStyles((theme) => ({
+  linkStyle: {
+    "&:hover": {
+      textDecoration: "underline",
+    },
+    color: "inherit",
+    fontSize: "1rem",
+    textDecoration: "none",
+  },
+  root: {
+    paddingTop: theme.spacing(3),
+    [theme.breakpoints.up("md")]: {
+      display: "flex",
+      maxWidth: theme.breakpoints.values.lg,
+      margin: "0 auto",
+      paddingTop: 0,
+    },
+  },
+}));
+
+const REQUEST_ID = "request";
+
+export default function UserPage() {
+  const classes = useStyles();
+  const history = useHistory();
+  let { username } =
+    useParams<{
+      username: string;
+    }>();
+
+  const { data: user, isLoading, error } = useUserByUsername(username, true);
+
+  const [isRequesting, setIsRequesting] = useState(false);
+  const [isSuccessRequest, setIsSuccessRequest] = useState(false);
+
+  useLayoutEffect(() => {
+    if (isRequesting) {
+      const requestEl = document.getElementById(REQUEST_ID);
+      requestEl?.scrollIntoView();
+    }
+  }, [isRequesting]);
+
+  return (
+    <>
+      {isSuccessRequest && (
+        <SuccessSnackbar>{SEND_REQUEST_SUCCESS}</SuccessSnackbar>
+      )}
+      {error && <Alert severity="error">{error}</Alert>}
+      {isLoading ? (
+        <CircularProgress />
+      ) : user ? (
+        <ProfileUserProvider user={user}>
+          <div className={classes.root}>
+            <Overview user={user} setIsRequesting={setIsRequesting} />
+            <UserCard
+              onTabChange={(newTab) => {
+                history.push(routeToUser(user.username, newTab));
+              }}
+              user={user}
```

Comment:
Although feel free to leave this in a separate PR/noted as a refactor task/issue if you like

Reply:
I'm not sure if I can remove the user prop here as the UserCard component is used both on ProfilePage and on UserPage. So the parent provided the user object here. One page will pick the user by username, the other page will provide the current authenticated user.

Reply:
It should be ok, since the same context can be reused in multiple places. React will know to pick the relevant one in the component tree, so the important thing is both `ProfilePage` and `UserPage` here wrap this part of the UI with the same `Provider` component.

Reply:
Ah I'm an idiot! Of course all that's needed is that the components are somewhere wrapped with the `ProfileUserProvider` and this is already the case on `ProfilePage` and `UserPage`.

Reply:
Hopefully I didn't overdo it now  🐰